### PR TITLE
Wall avoidance

### PR DESCRIPTION
**Pull request by johnny-la**
#### Wall Avoidance Steering Behaviour
* Added "Wall Avoidance" steering behaviour to `Steerable`
* Test this in _Scenes/AI Test/Obstacle Avoidance Example_
* Please do not use the "Obstacle Avoidance" steering behaviour. This only avoids circular obstacles. Wall avoidance allows NPCs to avoid obstacles of any shape. 

_Note:_ Obstacle and Wall Avoidance behaviours use the same properties in `SteeringTest.cs`.

### CODE REVIEW

**Comment by alisterdev** [comment]
image

- The flee guy seemed to be stuck in the wall
- 2 fishes once ran into each other

If the 2 things below are expected behaviour, then all is fine

File: Assets/Scripts/NPC/Steering Behaviours/Steerable.cs [hunk 1]
```diff
@@ -405,6 +405,63 @@ public Vector2 ObstacleAvoidanceForce(float avoidanceForce, float maxViewDistanc
...
+    /// Avoids the nearest obstacle in front of the object. This works for obstacles of any size or shape, unlike
+    /// Obstacle Avoidance, which approximates obstacles as spheres. Formal name: "Containment" or "Generalized Obstacle Avoidance"
+    /// </summary>
+    /// <param name="avoidanceForce">The amount of force applied in order to avoid the nearest obstacle.</param>
+	/// <param name="maxViewDistance">Only obstacles within 'maxViewDistance' meters of this steerable can be avoided.</param>
+	/// <param name="obstacleLayer">The layer which contains the colliders that can be avoided.</param>
+    private Vector2 WallAvoidanceForce(float avoidanceForce, float maxViewDistance, LayerMask obstacleLayer)
+    {
+        // Update the steerable's 'normalizedVelocity' variable
+		CheckDirty();
```
Comment (alisterdev) [inline Assets/Scripts/NPC/Steering Behaviours/Steerable.cs]: formating

File: Assets/Scripts/NPC/Steering Behaviours/SteeringTest.cs [hunk 2]
```diff
@@ -13,6 +13,7 @@ public enum SteeringBehaviorType
 	Evade,
 	Wander,
 	ObstacleAvoidance,
+    WallAvoidance,
```
Comment (alisterdev) [inline Assets/Scripts/NPC/Steering Behaviours/SteeringTest.cs]: formatting of the whole class

**Comment by johnny-la** [comment]
Fixed formatting. Fish running into each other is expected. *Separation* steering behaviour needs to be added to prevent this. 

The Flee fish running into a wall is unexpected, but fine at this point. It'll be a matter of playing with steering behaviour settings.

**Pull request merged**